src/main/js/utils/generate.js

```diff
@@ -76,17 +83,17 @@ const generateSchedules = (values, loadQuery = '') => (dispatch, getState) => {
           if (!gradeMap[name]) return 0;
 
           if (!gradeMap[name][instructor1] && !gradeMap[name][instructor2]) return 0;
-          if (!gradeMap[name][instructor1]) return 1;
-          if (!gradeMap[name][instructor2]) return -1;
+          if (!gradeMap[name][instructor1]) return gradeMap[name][instructor2] - gradeMap[name].Staff;
+          if (!gradeMap[name][instructor2]) return gradeMap[name].Staff - gradeMap[name].Staff;
```

Comment:
Like in `Schedule` you should use `useCourseAvg` instead of using this

Reply:
Done.